fdbserver/storageserver.actor.cpp

```diff
@@ -3020,6 +3021,11 @@ ACTOR Future<std::pair<ChangeFeedStreamReply, bool>> getChangeFeedMutations(Stor
 			decodedMutations.push_back(decodeChangeFeedDurableValue(kv.value));
 			if (doFilterMutations || !req.encrypted) {
 				for (auto& m : decodedMutations.back().first) {
+					if (g_network && g_network->isSimulated()) {
```

Comment:
any reason we want to limit the `ASSERT` to `simulation` environment; even in production such failure needs to be caught sooner...

Reply:
Same point here as well as other places in this PR.

Reply:
I think earlier everything was guarded under simulation for worries of cluster misconfiguration. I can change it everywhere